From 4564d0df2e704c05e8c992650ae2b7f7280e6e22 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Thu, 5 Jul 2018 08:56:11 +0200
Subject: [PATCH 026/137] fork: ipipe: announce mm dismantling

IPIPE_KEVT_CLEANUP is emitted before a process memory context is
entirely dropped, after all the mappings have been exited. Per-process
resources which might be maintained by the co-kernel could be released
there, as all tasks have exited.
---
 kernel/fork.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/kernel/fork.c b/kernel/fork.c
index dcab08c..599fbba 100644
--- a/kernel/fork.c
+++ b/kernel/fork.c
@@ -54,6 +54,7 @@
 #include <linux/futex.h>
 #include <linux/compat.h>
 #include <linux/kthread.h>
+#include <linux/ipipe.h>
 #include <linux/task_io_accounting_ops.h>
 #include <linux/rcupdate.h>
 #include <linux/ptrace.h>
@@ -926,6 +927,7 @@ static inline void __mmput(struct mm_struct *mm)
 	exit_aio(mm);
 	ksm_exit(mm);
 	khugepaged_exit(mm); /* must run before exit_mmap */
+	__ipipe_report_cleanup(mm);
 	exit_mmap(mm);
 	mm_put_huge_zero_page(mm);
 	set_mm_exe_file(mm, NULL);
-- 
1.9.1

